File: android/src/main/kotlin/com/gdelataillade/alarm/models/EditRingingAlarmSettings.kt
package com.gdelataillade.alarm.models

import com.gdelataillade.alarm.generated.EditRingingAlarmSettingsWire
import com.gdelataillade.alarm.generated.VoiceTagSettingsWire
import kotlinx.serialization.Serializable

@Serializable
data class EditRingingAlarmSettings(
    val id: Int,
    val volumeSettings: VolumeSettings?,
    val assetAudioPath: String?,
    val loopAudio: Boolean?,
    val vibrate: Boolean?,
    val flashlight: Boolean?,
    val voiceTagSettings: VoiceTagSettings?,
    val timePressureSettings: TimePressureSettings?,
) {
    companion object {
        fun fromWire(e: EditRingingAlarmSettingsWire): EditRingingAlarmSettings {
            return EditRingingAlarmSettings(
                e.id.toInt(),
                e.volumeSettings?.let { VolumeSettings.fromWire(it) },
                e.assetAudioPath,
                e.loopAudio,
                e.vibrate,
                e.flashlight,
                e.voiceTagSettings?.let { VoiceTagSettings.fromWire(it) },
                e.timePressureSettings?.let { TimePressureSettings.fromWire(it) },
            )
        }
    }
}

File: ios/Classes/models/EditRingingAlarmSettings.swift
import Foundation

struct EditRingingAlarmSettings {
    let id: Int
    let volumeSettings: VolumeSettings?
    let assetAudioPath: String?
    let loopAudio: Bool?
    let vibrate: Bool?
    let flashlight: Bool?
    let voiceTagSettings: VoiceTagSettings?
    let timePressureSettings: TimePressureSettings?

    static func from(wire: EditRingingAlarmSettingsWire) -> EditRingingAlarmSettings {
        return EditRingingAlarmSettings(
            id: Int(truncatingIfNeeded: wire.id),
            volumeSettings: wire.volumeSettings.map { VolumeSettings.from(wire: $0) },
            assetAudioPath: wire.assetAudioPath,
            loopAudio: wire.loopAudio,
            vibrate: wire.vibrate,
            flashlight: wire.flashlight,
            voiceTagSettings: wire.voiceTagSettings.map { VoiceTagSettings.from(wire: $0) },
            timePressureSettings: wire.timePressureSettings.map { TimePressureSettings.from(wire: $0) }
        )
    }
}
